tests/testthat/test-appMetadata-quarto.R

```diff
@@ -123,9 +99,42 @@ test_that("quartoInspect processes content with filenames containing spaces", {
   expect_equal(inspect$engines, c("markdown"))
 })
 
-test_that("quartoInspect returns NULL on non-quarto Quarto content", {
+test_that("quartoInspect produces an error when a document cannot be inspected", {
+  skip_if_no_quarto()
+
+  dir <- local_temp_app(list("bad.qmd" = c(
+    "---",
+    "format: unsupported",
+    "---",
+    "this is a document using an unsupported format."
+  )))
+  expect_error(
```

Comment:
I have a very mild preference for `expect_snapshot()` here so it's a bit easier to update if the error ever changes and so you can see if the correct call is bubbled up in the error.

Reply:
The challenge here is that Quarto emits a stack trace with source references. I'll use `expect_snapshot()` and throw a `transform` function into the mix.

Reply:
Done. Blindly `head(lines,3)` and using that as the important information.

Reply:
Also: `NOCOLOR=1` to suppress colors from Quarto error output.